src/cpp/session/modules/SessionDiagnostics.cpp

```diff
@@ -796,41 +805,50 @@ Error lintRSourceDocument(const json::JsonRpcRequest& request,
       return error;
    }
    
-   // Try to get the contents from the database
-   boost::shared_ptr<SourceDocument> pDoc(new SourceDocument());
-   error = get(documentId, pDoc);
-   
-   // don't log on error here (it's possible that we might attempt to lint a
-   // document immediately after a suspend-resume, and so we fail to get the
-   // contents of that document)
-   if (error)
-      return error;
-   
    FilePath origin = module_context::resolveAliasedPath(documentPath);
```

Comment:
Is this check still meaningful if we're linting an arbitrary chunk of content? Or should content always be thought of as a fragment of some already-existing document?

Reply:
It's helpful to always think of it as a fragment. It doesn't make a big difference here, though you could imagine that we had a linter that warned you if you referenced a path that didn't exist (relative to the document).